pkg/inject/inject.go

```diff
@@ -0,0 +1,595 @@
+	conf.proxyOutboundCapacity = m
+	return conf
+}
+
+// YamlMarshalObj returns the yaml for the workload in conf
+func (conf *ResourceConfig) YamlMarshalObj() ([]byte, error) {
+	return yaml.Marshal(conf.obj)
+}
+
+// ParseMetaAndYaml fills conf fields with both the metatada and the workload contents
+func (conf *ResourceConfig) ParseMetaAndYaml(bytes []byte) (*Report, error) {
+	if _, err := conf.ParseMeta(bytes); err != nil {
+		return nil, err
+	}
+	r := newReport(conf)
+	return &r, conf.parse(bytes)
+}
+
+// ParseMeta extracts metadata from bytes.
+// It returns false if the workload's payload is empty
+func (conf *ResourceConfig) ParseMeta(bytes []byte) (bool, error) {
+	if err := yaml.Unmarshal(bytes, &conf.meta); err != nil {
+		return false, err
+	}
+	if err := yaml.Unmarshal(bytes, &conf.objMeta); err != nil {
+		return false, err
+	}
+	return conf.objMeta.ObjectMeta != nil, nil
+}
+
+// GetPatch returns the JSON patch containing the proxy and init containers specs, if any
+func (conf *ResourceConfig) GetPatch(bytes []byte) ([]byte, []Report, error) {
+	report := newReport(conf)
+	log.Infof("working on %s %s..", strings.ToLower(conf.meta.Kind), conf.objMeta.Name)
+
+	if err := conf.parse(bytes); err != nil {
+		return nil, []Report{report}, err
+	}
+
+	var patch *Patch
+	if strings.ToLower(conf.meta.Kind) == k8s.Pod {
+		patch = NewPatchPod()
+	} else {
+		patch = NewPatchDeployment()
+	}
+
+	// If we don't inject anything into the pod template then output the
+	// original serialization of the original object. Otherwise, output the
+	// serialization of the modified object.
+	if conf.podSpec != nil {
+		metaAccessor, err := k8sMeta.Accessor(conf.obj)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		// The namespace isn't necessarily in the input so it has to be substituted
+		// at runtime. The proxy recognizes the "$NAME" syntax for this variable
+		// but not necessarily other variables.
+		identity := k8s.TLSIdentity{
+			Name:                metaAccessor.GetName(),
+			Kind:                strings.ToLower(conf.meta.Kind),
+			Namespace:           "$" + podNamespaceEnvVarName,
+			ControllerNamespace: conf.globalConfig.GetLinkerdNamespace(),
+		}
+
+		report.update(conf)
+		if conf.shouldInject(report) {
+			conf.injectPodSpec(patch, identity)
+			conf.injectObjectMeta(patch)
+		}
+	} else {
+		report.UnsupportedResource = true
+	}
+
+	patchJSON, err := json.Marshal(patch.patchOps)
+	log.Debugf("patch: %s", patchJSON)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return patchJSON, []Report{report}, nil
+}
+
+// KindInjectable returns true if the resource in conf can be injected with a proxy
+func (conf *ResourceConfig) KindInjectable() bool {
+	for _, kind := range injectableKinds {
+		if strings.ToLower(conf.meta.Kind) == kind {
+			return true
+		}
+	}
+	return false
+}
+
+func (conf *ResourceConfig) getFreshWorkloadObj() runtime.Object {
```

Comment:
mind adding a comment that the objects supported here should match `injectableKinds`?